sdk/iot/Azure.ResourceManager.IotFirmwareDefense/src/Customization/Models/BinaryHardeningResult.cs

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using Azure.Core;
+using Azure.ResourceManager.Models;
+
+namespace Azure.ResourceManager.IotFirmwareDefense.Models
+{
+    /// <summary> binary hardening analysis result resource. </summary>
+    public partial class BinaryHardeningResult
+    {
+        /// <summary> Initializes a new instance of <see cref="BinaryHardeningResult"/>. </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public BinaryHardeningResult()
+        {
+        }
+        /// <summary>
+        /// The architecture of the binary being reported on.
+        /// Serialized Name: BinaryHardeningResource.properties.architecture
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public string Architecture
+        {
+            get => ExecutableArchitecture;
+            set => ExecutableArchitecture = value;
+        }
+        /// <summary>
+        /// The executable class to indicate 32 or 64 bit.
+        /// Serialized Name: BinaryHardeningResource.properties.class
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public string Class
+        {
+            get => ExecutableClass.ToString();
+            set => ExecutableClass = value;
+        }
+        /// <summary> NX (no-execute) flag. </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public bool? NXFlag
+        {
+            get => SecurityHardeningFeatures.NoExecute;
+            set => SecurityHardeningFeatures.NoExecute = value;
```

Comment:
I think we need null checks
```suggestion
            get => SecurityHardeningFeatures?.NoExecute;
            set 
            {
            	SecurityHardeningFeatures ??= new();
            	SecurityHardeningFeatures.NoExecute = value;
            }
```